### DISCUSSION TITLE
Branch protection pattern only matching branches authored by me

### DISCUSSION THREAD

**Discussion by gmulz** [community/community#57024] [Repositories]
### Select Topic Area

Bug

### Body

I am attempting to apply a branch protection rule to all branches in my repository, using the pattern `*`. When I save this rule, I see the "Currently applies to 19 branches". The branches listed that the rules are applied to are only branches authored by me (repo admin), but none of the other branches authored by other contributors in the repo. There are 33 total branches in the repo that the rule should be applied to.

When I look at the list of branches, I see a small "view rules" badge next to only my branches, none of the other ones. Clicking on this takes me to the Rulesets page, where I see "No rulesets have been added yet". This seems like a bug with the new Rulesets implementation colliding with the branch protections, but I have no idea why it's only applying to my branches and not the rest of the branches in the repo. 

<img width="93" alt="image" src="https://github.com/community/community/assets/6372359/c693fc48-3f51-4fdd-8c16-529c2aa83770">
<img width="775" alt="image" src="https://github.com/community/community/assets/6372359/3913a4e4-b078-4f4f-a628-14173a445ee2">

**Comment by ikozigbo** [comment]


**Reply by gravidThoughts** [reply]
Did you read the question? He is asking why his branch protection definition doesn't apply to all branches in the repo, regardless of the author, as he expects it will. Not to demean your excellent guidance on how to do what he doesn't want to do, but how is it possible that this was upvoted?

**Reply by tuves** [reply]
Hi @ikozigbo, 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by patrick-knight** [comment]
Hey @gmulz the link on the 🛡️ there only references created rulesets in the repo and will not show any existing branch protections. This is expected behavior as we intend to have rules eventually be the replacement for branch protections.
